include/claws/iterator_util.hpp

```diff
@@ -10,7 +10,7 @@ namespace claws
 
   public:
     using difference_type = typename std::iterator_traits<raw_it_type>::difference_type;
-    using value_type = std::remove_cv_t<decltype(func(*it))>;
+    using value_type = std::remove_reference_t<decltype(func(*it))>;
```

Comment:
Generally we prefer use both remove_reference and remove_cv_t qualifier if you realy want a value, usualy it's resumed to decay to a real type: https://en.cppreference.com/w/cpp/types/decay

Reply:
I would agree in the general sense but I'm not sure I want the iterator to return a non-const value if the functor does. So I wouldn't use `std::remove_cv` in this case.